### PR TITLE
Add chip-specific features for tests/examples

### PR DESCRIPTION
**Pull request by bugadani**
[Motivator](https://github.com/esp-rs/esp-hal/pull/2161#discussion_r1761600859)

### CODE REVIEW

File: xtask/src/lib.rs [hunk 1]
```diff
@@ -161,38 +155,56 @@ pub fn load_examples(path: &Path, action: CargoAction) -> Result<Vec<Metadata>>
...
-                bail!(
-                    "Expected at least two elements (key, value), found {}",
-                    split.len()
-                );
-            }
+            let Some((key, value)) = line.trim_start_matches("//%").split_once(':') else {
+                bail!("Metadata line is missing ':': {}", line);
+            };
 
-            // The trailing ':' on metadata keys is optional :)
```
Comment (bjoernQ) [inline xtask/src/lib.rs:183]: technically a change but given I didn't even knew this I guess that is totally fine

File: xtask/src/lib.rs [hunk 2]
```diff
@@ -161,38 +155,56 @@ pub fn load_examples(path: &Path, action: CargoAction) -> Result<Vec<Metadata>>
...
+                let values = value
+                    .split_ascii_whitespace()
+                    .map(ToString::to_string)
+                    .collect::<Vec<_>>();
+
+                for chip in chips {
+                    chip_features.insert(chip, values.clone());
+                }
             } else {
                 log::warn!("Unrecognized metadata key '{key}', ignoring");
```
Comment (bjoernQ) [inline xtask/src/lib.rs:209]: unrelated but we might consider to make this an hard error and panic - took we a while to recognize I was testing with `CHIP_FEATURE(esp32s3)` 😄

**Review by bjoernQ — APPROVED** [review]
Works fine - not sure we should mention this in the README since it's not yet used anywhere (so it's hard to discover)

**Review by JurajSadel — APPROVED** [review]
Thanks!

**Pull request merged**